File: 0729-my-calendar-i/0729-my-calendar-i.py
class MyCalendar:

    def __init__(self):
        self.bookings = []  # Stores booked intervals

    def book(self, start: int, end: int) -> bool:
        for s, e in self.bookings:
            # Check for overlap
            if max(start, s) < min(end, e):
                return False
        # Add the interval if no overlap is found
        self.bookings.append((start, end))
        return True

File: 0729-my-calendar-i/0729-my-calendar-i.go

type MyCalendar struct {
	bookings [][2]int // Stores booked intervals
}

func Constructor() MyCalendar {
	return MyCalendar{bookings: make([][2]int, 0)}
}

func (this *MyCalendar) Book(start int, end int) bool {
	for _, interval := range this.bookings {
		// Check for overlap
		if max(start, interval[0]) < min(end, interval[1]) {
			return false
		}
	}
	// Add the interval if no overlap is found
	this.bookings = append(this.bookings, [2]int{start, end})
	return true
}